### PR TITLE
8295350: JFR: Add stop methods for recording streams

### PR DESCRIPTION
**Pull request by egahlin**
Could I have a review of PR that adds stop() methods to RecordingStream and RemoteRecordingStream. Purpose is to provide better control of what is being recorded.

Testing: test/jdk/jdk/jfr

Thanks
Erik

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8295350](https://bugs.openjdk.org/browse/JDK-8295350): JFR: Add stop methods for recording streams
 * [JDK-8295487](https://bugs.openjdk.org/browse/JDK-8295487): JFR: Add stop methods for recording streams (**CSR**)

### Reviewers
 * [Markus Grönlund](https://openjdk.org/census#mgronlun) (@mgronlun - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11337/head:pull/11337` \
`$ git checkout pull/11337`

Update a local copy of the PR: \
`$ git checkout pull/11337` \
`$ git pull https://git.openjdk.org/jdk pull/11337/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11337`

View PR using the GUI difftool: \
`$ git pr show -t 11337`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11337.diff">https://git.openjdk.org/jdk/pull/11337.diff</a>

</details>

### CODE REVIEW

File: src/jdk.jfr/share/classes/jdk/jfr/consumer/RecordingStream.java [hunk 1]
```diff
@@ -380,6 +381,41 @@ public void startAsync() {
...
+     * conjunction with the {@link #startAsync()} method to monitor what happens
+     * during a test method:
+     * <p>
+     * {@snippet class="Snippets" region="RecordingStreamStop"}
+     *
+     * @return {@code true} if recording is stopped, {@code false} otherwise
+     *
+     * @throws IllegalStateException if the recording is not started or is already stopped
+     */
+    public boolean stop() {
```
Comment (AlanBateman) [inline src/jdk.jfr/share/classes/jdk/jfr/consumer/RecordingStream.java:404]: Drive-by comment, I assume you should add `@since 20` here.

File: src/jdk.management.jfr/share/classes/jdk/management/jfr/RemoteRecordingStream.java [hunk 2]
```diff
@@ -560,6 +561,75 @@ public void startAsync() {
...
+            if (!started) {
+                throw new IllegalStateException("Event stream must be started before it can stopped");
+            }
+            try {
+                boolean stopped = false;
+                try (StreamBarrier pb = ManagementSupport.activateStreamBarrier(stream)) {
+                    try (StreamBarrier rb = repository.activateStreamBarrier()) {
+                        stopped = mbean.stopRecording(recordingId);
+                        ManagementSupport.setCloseOnComplete(stream, false);
+                        long stopTime = getRecordingInfo(mbean.getRecordings(), recordingId).getStopTime();                       pb.setStreamEnd(stopTime);
```
Comment (turbanoff) [inline src/jdk.management.jfr/share/classes/jdk/management/jfr/RemoteRecordingStream.java:615]: Why `pb.setStreamEnd(stopTime);` is placed at the end of line here?

Comment (egahlin) [reply src/jdk.management.jfr/share/classes/jdk/management/jfr/RemoteRecordingStream.java:615]: I wondered why "pb.setStreamEnd(stopTime);" disappeared. I will remove it.

**Review by mgronlun — APPROVED** [review]
